be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -99,16 +99,26 @@ Status HdfsScannerCSVReader::_fill_buffer() {
 Status HdfsTextScanner::do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) {
     TTextFileDesc text_file_desc = _scanner_params.scan_ranges[0]->text_file_desc;
     _field_delimiter = text_file_desc.field_delim;
-    // we should cast string to char now since csv reader only support record delimiter by char
+
+    // we should cast string to char now since csv reader only support record, collection, mapkey delimiter by char
     _record_delimiter = text_file_desc.line_delim.front();
+    _collection_delimiter = text_file_desc.collection_delim.front();
+    _mapkey_delimiter = text_file_desc.mapkey_delim.front();
+
+    // _collection_delimiter and _mapkey_delimiter won't be empty
+    DCHECK(!_collection_delimiter.empty());
+    DCHECK(!_mapkey_delimiter.empty());
     return Status::OK();
 }
 
 Status HdfsTextScanner::do_open(RuntimeState* runtime_state) {
     RETURN_IF_ERROR(_create_or_reinit_reader());
     SCOPED_RAW_TIMER(&_stats.reader_init_ns);
-    for (int i = 0; i < _scanner_params.materialize_slots.size(); i++) {
-        auto slot = _scanner_params.materialize_slots[i];
+    for (const auto& slot : _scanner_params.materialize_slots) {
```

Comment:
If slot is a raw pointer, needn't use a reference.
use `cont auto slot` directly.